Formula/mercurial.rb

```diff
@@ -13,6 +14,14 @@ class Mercurial < Formula
     sha256 "8934a8fc26c36e56f017d99ac43a0ebe9d6922f8276c1dfccd9ccb980ae05e45" => :mavericks
   end
 
+  option "with-custom-python", "Install against the python in PATH instead of "\
+                               "Homebrew's python"
+  if build.with? "custom-python"
```

Comment:
We want different dependencies depending on an option that doesn't fit the :recommended form so I think that's going to be hard to swing; open to ideas.